diff --git a/src/custom_controlnet_aux/depth_anything_v2/dpt.py b/src/custom_controlnet_aux/depth_anything_v2/dpt.py
index 630ae69..9b69aed 100644
--- a/src/custom_controlnet_aux/depth_anything_v2/dpt.py
+++ b/src/custom_controlnet_aux/depth_anything_v2/dpt.py
@@ -185,7 +185,8 @@ class DepthAnythingV2(nn.Module):
     @torch.no_grad()
     def infer_image(self, raw_image, input_size=518, max_depth=20.0):
         image, (h, w) = self.image2tensor(raw_image, input_size)
-        
+        image = image.to(next(self.parameters()).device)
+
         depth = self.forward(image, max_depth)
         
         depth = F.interpolate(depth[:, None], (h, w), mode="bilinear", align_corners=True)[0, 0]
@@ -214,7 +215,7 @@ class DepthAnythingV2(nn.Module):
         image = transform({'image': image})['image']
         image = torch.from_numpy(image).unsqueeze(0)
         
-        DEVICE = 'cuda' if torch.cuda.is_available() else 'mps' if torch.backends.mps.is_available() else 'cpu'
+        DEVICE = next(self.parameters()).device
         image = image.to(DEVICE)
         
-        return image, (h, w)
\ No newline at end of file
+        return image, (h, w)
